src/mlte/measurement/external_measurement.py

```diff
@@ -4,32 +4,59 @@
 
 from __future__ import annotations
 
-from typing import Type
+from typing import Type, Callable, Optional
+import typing
 
 from mlte.value import Value
 from .measurement import Measurement
 
 
 class ExternalMeasurement(Measurement):
-    def __init__(self, identifier: str, value_type: type):
+    @typing.no_type_check
+    def __init__(
+        self,
+        identifier: str,
+        value_type: type,
+        function: Optional[Callable] = None,
+    ):
         """
         Initialize a new ExternalMeasurement measurement.
 
         :param identifier: A unique identifier for the instance
         :type identifier: str
         :param value_type: The type of the Value this measurement will return.
         :type value_type: Type
+        :param function: The function to be used when evaluating.
+        :type function: Optional[Callable]
         """
         super().__init__(self, identifier)
+
         if not issubclass(Value, value_type):
             raise Exception(
-                f"Value type provided is not a subtype of Value: {self.value_type}"
+                f"Value type provided is not a subtype of Value: {value_type}"
             )
         self.value_type: type = value_type
 
+        if function is not None:
+            if not callable(function):
```

Comment:
Info: this is a cool function, I didn't know this was a builtin.

Reply:
Just learned it myself :)